test/parallel/test-async-wrap-GH13045.js

```diff
@@ -0,0 +1,63 @@
+'use strict';
+const common = require('../common');
+
+// checks for the issue in https://github.com/nodejs/node/issues/13045
+// An Http Agent reuses a TLSSocket, fails to call `asyncReset`
+
+const assert = require('assert');
+const https = require('https');
+const fs = require('fs');
+
+const serverOptions = {
+  key: fs.readFileSync(`${common.fixturesDir}/keys/agent1-key.pem`),
+  cert: fs.readFileSync(`${common.fixturesDir}/keys/agent1-cert.pem`),
+  ca: fs.readFileSync(`${common.fixturesDir}/keys/ca1-cert.pem`)
+};
+
+const server = https.createServer(serverOptions, (req, res) => {
+  res.end('hello world\n');
+});
+
+server.listen(0, common.localhostIPv4, common.mustCall(function () {
+  const port = this.address().port;
+  const clientOptions = {
+    agent: new https.Agent({
+      keepAlive: true,
+      maxSockets: 50,
+      rejectUnauthorized: false,
+    }),
+    hostname: common.localhostIPv4,
+    port: port,
+    path: '/',
+    method: 'GET'
+  };
+
+  const req = https.request(clientOptions, common.mustCall((res) => {
+    assert.strictEqual(res.statusCode, 200);
+    res.on('error', common.mustNotCall());
```

Comment:
Is there a reason to add this listener? If there is an error in the future I think it would be better to see that error than an `AssertionError('must now throw')` message.